### PR TITLE
Proposal for error-handling

### PR DESCRIPTION
**Pull request by ineiti**
Is this OK like that, or should we keep the "shouldn't|couldn't|wouldn't|hasn't"-format?

### CODE REVIEW

**Comment by nkcr** [comment]
> Is this OK like that, or should we keep the "shouldn't|couldn't|wouldn't|hasn't"-format?

IMHO errors should never be thrown without a message: it helps reading the errors (obviously) and the source code. So we could add `Errors are systematically wrapped using xerrors AND a meaningful message`.

**Comment by tharvik** [comment]
Might want to also update `.github/PULL_REQUEST_TEMPLATE.md`

**Comment by ineiti** [comment]
> Might want to also update `.github/PULL_REQUEST_TEMPLATE.md`

That's the change I'm proposing ;)

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -11,7 +11,7 @@ This PR
 - [ ] 1. Design choices, user documentation and any additional doc are added (or updated) in READMEs.
 - [ ] 2. Any new behaviour is tested and small units of code that can be are unit tested.
 - [ ] 3. Code comments are added on tests to explain what they do.
-- [ ] 4. Errors are systematically wrapped, following the `[failed to | couldn't do] ACTION THAT FAILED: " + err.Error()` format.
+- [ ] 4. Errors are systematically wrapped using `xerrors` AND a meaningful message is added.
```
Comment (nkcr) [inline .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
- [ ] 4. Errors are systematically wrapped with a meaningful message using `xerrors.Errorf` and the `%+v` verb.
```

**Pull request merged**